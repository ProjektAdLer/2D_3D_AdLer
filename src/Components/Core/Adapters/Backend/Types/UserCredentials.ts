type UserCredentials = {
  username: string;
  password: string;
};

export default UserCredentials;
