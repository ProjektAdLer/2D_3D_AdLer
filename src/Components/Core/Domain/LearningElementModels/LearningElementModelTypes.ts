enum TextElementModelTypes {
  //default
  Bookshelf1 = "l_text_bookshelf_1",
  Bookshelf2 = "l_text_bookshelf_2",
  //arcade
  ComicShelfbig = "l_text_comicshelfbig_1",
  ComicShelfsmall = "l_text_comicshelfsmall_1",
  //campus
  LibraryShelf = "l_text_libraryshelf_1",
}

enum ImageElementModelTypes {
  //default
  Painting1 = "l_image_painting_1",
  Painting2 = "l_image_painting_2",
  PaintingEasel = "l_image_paintingeasel_1",
  //arcade
  Gameposter1 = "l_image_gameposter_1",
  Gameposter2 = "l_image_gameposter_2",
  Cardboardcutout = "l_image_cardboardcutout_1",
  //campus
  sciencegeo1 = "l_image_sciencegeo_1",
  sciencebio1 = "l_image_sciencebio_1",
  sciencewhiteboard = "l_image_sciencewhiteboard_1",
}

enum VideoElementModelTypes {
  //default
  Television = "l_video_television_1",
  //arcade
  vrdesk = "l_video_vrdesk_1",
  //campus
  movieprojector = "l_video_movieprojector_1",
}

// enum PdfElementModelTypes {}

enum H5pElementModelTypes {
  //default
  Blackboard = "l_h5p_blackboard_1",
  DeskPC1 = "l_h5p_deskpc_1",
  DrawingTable = "l_h5p_drawingtable_1",
  SlotMachine = "l_h5p_slotmachine_1",
  //arcade
  GreySlotmachine = "l_h5p_greyslotmachine_1",
  RedSlotmachine = "l_h5p_redslotmachine_1",
  PurpleSlotmachine = "l_h5p_purpleslotmachine_1",
  BlackSlotmachine = "l_h5p_blackslotmachine_1",
  DeskPC2 = "l_h5p_deskpc_2",
  //campus
  Blackboard2 = "l_h5p_blackboard_2",
  DeskPC3 = "l_h5p_deskpc_3",
  DrawingTable2 = "l_h5p_drawingtable_2",
  Daylightprojector = "l_h5p_daylightprojector_1",
}

enum QuizElementModelTypes {
  RobotNPC = "a_npc_alerobot",
  DefaultNPC = "a_npc_defaultnpc",
  ArcadeNPC = "a_npc_sheriffjustice",
  CampusNPC = "a_npc_dozentlukas",
}
enum TrophyElementModelTypes {
  BronzeTrophy = "l_trophy_bronze",
  SilverTrophy = "l_trophy_silver",
  GoldTrophy = "l_trophy_gold",
}

// enum NoElementModelTypes {
//   None = "",
// }

/** 
 Pseudo-nested enum to reference all possible members of the element model enums.
 */
export const LearningElementModelTypeEnums = {
  TextElementModelTypes: TextElementModelTypes,
  ImageElementModelTypes: ImageElementModelTypes,
  VideoElementModelTypes: VideoElementModelTypes,
  // PdfElementModelTypes: PdfElementModelTypes,
  H5pElementModelTypes: H5pElementModelTypes,
  QuizElementModelTypes: QuizElementModelTypes,
  TrophyElementModelTypes: TrophyElementModelTypes,
  // NoElementModelTypes: NoElementModelTypes,
};

export function isValidLearningElementModelType(type: string): boolean {
  return Object.values(LearningElementModelTypeEnums).some((enumType) =>
    Object.values(enumType).includes(type)
  );
}

/**
 Type of a learning element model.
  */
export type LearningElementModel =
  | TextElementModelTypes
  | ImageElementModelTypes
  | VideoElementModelTypes
  // | PdfElementModelTypes
  | H5pElementModelTypes
  | QuizElementModelTypes
  | TrophyElementModelTypes;
// | NoElementModelTypes;
