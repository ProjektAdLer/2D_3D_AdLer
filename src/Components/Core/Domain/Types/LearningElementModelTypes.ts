enum TextElementModelTypes {
  Bookshelf1 = "l_text_bookshelf_1",
  Bookshelf2 = "l_text_bookshelf_2",
}

enum ImageElementModelTypes {
  // TODO: rename to l_image_... when models are updated
  Painting1 = "l_picture_painting_1",
  Painting2 = "l_picture_painting_2",
  PaintingEasel = "l_picture_paintingeasel_1",
}

enum VideoElementModelTypes {
  Television = "l_video_television_1",
}

enum PdfElementModelTypes {}

enum H5pElementModelTypes {
  Backboard = "l_h5p_backboard_1",
  DeskPC = "l_h5p_deskpc_1",
  DrawingTable = "l_h5p_drawingtable_1",
  SlotMachine = "l_h5p_slotmachine_1",
}

enum NoElementModelTypes {
  None = "",
}

/** 
 Pseudo-nested enum to reference all possible members of the element model enums.
 */
export const LearningElementModelTypeEnums = {
  TextElementModelTypes: TextElementModelTypes,
  ImageElementModelTypes: ImageElementModelTypes,
  VideoElementModelTypes: VideoElementModelTypes,
  PdfElementModelTypes: PdfElementModelTypes,
  H5pElementModelTypes: H5pElementModelTypes,
  NoElementModelTypes: NoElementModelTypes,
};

/**
 Type of a learning element model.
  */
export type LearningElementModel =
  | TextElementModelTypes
  | ImageElementModelTypes
  | VideoElementModelTypes
  | PdfElementModelTypes
  | H5pElementModelTypes
  | NoElementModelTypes;
