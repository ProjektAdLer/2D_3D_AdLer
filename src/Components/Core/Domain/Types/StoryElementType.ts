/** binary encoded:
 * 00 - none
 * 01 - intro
 * 10 - outro
 * 11 - intro and outro
 */
export enum StoryElementType {
  None = 0, // 00
  Intro = 1, // 01
  Outro = 2, // 10
  IntroOutro = 3, // 11
}
