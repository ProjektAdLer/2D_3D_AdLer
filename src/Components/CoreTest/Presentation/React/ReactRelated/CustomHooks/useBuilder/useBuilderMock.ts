import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
