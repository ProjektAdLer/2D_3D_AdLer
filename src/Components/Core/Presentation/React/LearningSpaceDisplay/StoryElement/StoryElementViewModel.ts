import { StoryElementType } from "src/Components/Core/Domain/Types/StoryElementType";
import Observable from "../../../../../../Lib/Observable";
import { LearningElementModel } from "src/Components/Core/Domain/LearningElementModels/LearningElementModelTypes";

export default class StoryElementViewModel {
  introTexts: Observable<string[]> = new Observable<string[]>();
  outroTexts: Observable<string[]> = new Observable<string[]>();
  isOpen: Observable<boolean> = new Observable<boolean>(false);
  pageId: Observable<number> = new Observable<number>(0);
  outroUnlocked: Observable<boolean> = new Observable<boolean>(false);
  outroJustNowUnlocked: Observable<boolean> = new Observable<boolean>(false);

  //MenuNavigation
  showOnlyIntro: Observable<boolean> = new Observable<boolean>(false);
  showOnlyOutro: Observable<boolean> = new Observable<boolean>(false);

  //Story Types
  type: Observable<StoryElementType[]> = new Observable<StoryElementType[]>([]);
  modelType: Observable<LearningElementModel[]> = new Observable<
    LearningElementModel[]
  >([]);

  isSplitStory: Observable<boolean> = new Observable<boolean>(false);
  pickedStory: Observable<StoryElementType> =
    new Observable<StoryElementType>();
}
