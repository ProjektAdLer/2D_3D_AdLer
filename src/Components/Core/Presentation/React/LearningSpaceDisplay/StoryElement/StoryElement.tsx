import closeIcon from "../../../../../../Assets/icons/close.svg";
import { AdLerUIComponent } from "src/Components/Core/Types/ReactTypes";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import StoryElementViewModel from "./StoryElementViewModel";
import IStoryElementController from "./IStoryElementController";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import useObservable from "~ReactComponents/ReactRelated/CustomHooks/useObservable";
import { useTranslation } from "react-i18next";
import tailwindMerge from "../../../Utils/TailwindMerge";
import { StoryElementType } from "src/Components/Core/Domain/Types/StoryElementType";
import StyledContainer from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledContainer";
import StorySelection from "./StorySelection";
import SingleStoryLayout from "./SingleStoryLayout";
import { useCallback } from "react";
import CloseButton from "~ReactComponents/ReactRelated/ReactBaseComponents/CloseButton";
import { getNPCImage } from "../../../Utils/GetNPCImage";

export default function StoryElement({ className }: AdLerUIComponent<{}>) {
  const [viewModel, controller] = useBuilder<
    StoryElementViewModel,
    IStoryElementController
  >(BUILDER_TYPES.IStoryElementBuilder);
  const [isOpen] = useObservable<boolean>(viewModel?.isOpen);
  const [storyTypeToDisplay] = useObservable<StoryElementType>(
    viewModel?.storyTypeToDisplay,
  );
  const [outroUnlocked] = useObservable<boolean>(viewModel?.isOutroUnlocked);
  const [introCutsceneRunning] = useObservable<boolean>(
    viewModel?.isIntroCutsceneRunning,
  );
  const [outroCutsceneRunning] = useObservable<boolean>(
    viewModel?.isOutroCutsceneRunning,
  );
  const [introModelType] = useObservable(viewModel?.introModelType);
  const [outroModelType] = useObservable(viewModel?.outroModelType);

  const [introEmotion] = useObservable(viewModel?.introEmotion);
  const [outroEmotion] = useObservable(viewModel?.outroEmotion);

  const { t: translate } = useTranslation("learningSpace");

  const getTitleText = useCallback(() => {
    switch (storyTypeToDisplay) {
      case StoryElementType.Intro:
        return translate("introStoryTitle").toString();
      case StoryElementType.Outro:
        return translate("outroStoryTitle").toString();
      case StoryElementType.IntroOutro:
        return translate("introOutroStoryTitle").toString();
      case StoryElementType.None:
      default:
        return "";
    }
  }, [storyTypeToDisplay, translate]);

  const getContentTexts = useCallback(() => {
    switch (storyTypeToDisplay) {
      case StoryElementType.Intro:
        return viewModel.introTexts.Value;
      case StoryElementType.Outro:
        if (outroUnlocked) return viewModel.outroTexts.Value;
        else return [translate("outroLockedText").toString()];
      default:
        return [];
    }
  }, [viewModel, storyTypeToDisplay, outroUnlocked, translate]);

  if (!viewModel || !controller) return null;
  if (!isOpen || storyTypeToDisplay === StoryElementType.None) return null;

  return (
    <StyledContainer className={tailwindMerge(className, "")}>
      <div className="fixed top-0 bottom-0 left-0 right-0 z-50 flex flex-col items-center justify-center w-screen h-full bg-blacktrans lg:grid lg:grid-rows-3 lg:items-start">
        {/* Desktop NPC Thumbnail */}
        <div className="relative flex items-end justify-start invisible w-full row-start-2 pl-16 lg:visible lg:h-full">
          <div className="absolute bottom-0 z-50 w-8 h-16 left-44 bg-buttonbgblue bubblecornertopleft"></div>
          <img
            className="z-20 invisible object-contain h-0 -scale-x-100 brightness-125 lg:visible lg:h-full "
            alt="LearningImage!"
            data-testid="npcImage"
            src={getNPCImage(
              (storyTypeToDisplay & StoryElementType.Intro) ===
                StoryElementType.Intro
                ? introModelType
                : outroModelType,
              true,
              (storyTypeToDisplay & StoryElementType.Intro) ===
                StoryElementType.Intro
                ? introEmotion
                : outroEmotion,
            )}
          />
        </div>

        {/* Modal */}
        <div className="z-50 flex items-start justify-center pb-2 w-full lg:w-[95vw] max-w-7xl lg:max-h-[30vh] xl:max-h-[32vh] pt-2 lg:pt-0 row-start-3 overflow-auto">
          <div className="flex flex-col p-2 xl:px-8 gap-2 justify-center rounded-lg bg-buttonbgblue h-full w-full max-w-[95%]">
            {/* Header */}
            <div className="z-20 flex items-start justify-center gap-2 pt-2 overflow-hidden text-xl font-bold lg:max-w-5xl xl:max-w-6xl text-adlerdarkblue lg:roboto-black lg:text-2xl ">
              {/* Mobile NPC Thumbnail */}
              <img
                className="visible h-16 -scale-x-100 lg:invisible lg:h-0"
                alt="LearningImage!"
                date-testid="npcImage"
                src={getNPCImage(
                  (storyTypeToDisplay & StoryElementType.Intro) ===
                    StoryElementType.Intro
                    ? introModelType
                    : outroModelType,
                  false,
                  (storyTypeToDisplay & StoryElementType.Intro) ===
                    StoryElementType.Intro
                    ? introEmotion
                    : outroEmotion,
                )}
              />

              {/* Title */}
              <div className="w-full lg:text-lg">
                {storyTypeToDisplay & StoryElementType.Intro
                  ? (viewModel.introModelName.Value ?? getTitleText())
                  : (viewModel.outroModelName.Value ?? getTitleText())}
              </div>

              {/* Close Button */}
              <CloseButton
                onClick={() => controller.closePanel()}
                className="w-8 h-8 p-1 text-xs roboto-black xl:w-10 xl:h-10 lg:w-10 lg:h-10 md:w-10 md:h-10 sm:w-10 sm:h-10 lg:text-lg"
                title={translate("closeStoryToolTip").toString()}
              >
                <img
                  src={closeIcon}
                  className="lg:w-10 md:w-8 sm:w-6"
                  alt="CloseButton"
                />
              </CloseButton>
            </div>

            {/* Content */}
            {storyTypeToDisplay === StoryElementType.IntroOutro && (
              <StorySelection controller={controller} />
            )}

            {(storyTypeToDisplay === StoryElementType.Intro ||
              storyTypeToDisplay === StoryElementType.Outro) && (
              <SingleStoryLayout
                contentTexts={getContentTexts()}
                controller={controller}
                withBackButton={
                  !introCutsceneRunning &&
                  outroUnlocked &&
                  !outroCutsceneRunning &&
                  introModelType !== undefined &&
                  outroModelType !== undefined &&
                  introModelType === outroModelType
                }
              />
            )}
          </div>
        </div>
      </div>
    </StyledContainer>
  );
}
