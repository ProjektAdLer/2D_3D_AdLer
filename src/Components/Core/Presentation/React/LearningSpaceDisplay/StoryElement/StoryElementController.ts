import bind from "bind-decorator";
import IStoryElementController from "./IStoryElementController";
import StoryElementViewModel from "./StoryElementViewModel";
import CoreDIContainer from "~DependencyInjection/CoreDIContainer";
import IEndStoryElementCutScene from "src/Components/Core/Application/UseCases/EndStoryElementCutScene/IEndStoryElementCutSceneUseCase";
import USECASE_TYPES from "~DependencyInjection/UseCases/USECASE_TYPES";
import { StoryElementType } from "src/Components/Core/Domain/Types/StoryElementType";
import IStoryNPCPresenter from "../../../Babylon/StoryNPC/IStoryNPCPresenter";
import PRESENTATION_TYPES from "~DependencyInjection/Presentation/PRESENTATION_TYPES";

export default class StoryElementController implements IStoryElementController {
  private storyNPCPresenter: IStoryNPCPresenter;

  constructor(private viewModel: StoryElementViewModel) {}

  @bind
  closePanel(): void {
    this.viewModel.isOpen.Value = false;

    if (!this.storyNPCPresenter) {
      // no initialization in constructor, because storynpc is build after storyelement ~sb
      this.storyNPCPresenter = CoreDIContainer.get<IStoryNPCPresenter>(
        PRESENTATION_TYPES.IStoryNPCPresenter
      );
    }
    this.storyNPCPresenter.changeStateToRandomMovement();

    if (
      this.viewModel.isIntroCutsceneRunning.Value ||
      this.viewModel.isOutroCutsceneRunning.Value
    ) {
      this.viewModel.isOutroCutsceneRunning.Value = false;
      this.viewModel.isIntroCutsceneRunning.Value = false;

      CoreDIContainer.get<IEndStoryElementCutScene>(
        USECASE_TYPES.IEndStoryElementCutSceneUseCase
      ).execute();
    }
  }

  @bind
  onIntroButtonClicked(): void {
    this.viewModel.storyTypeToDisplay.Value = StoryElementType.Intro;
  }

  @bind
  onOutroButtonClicked(): void {
    this.viewModel.storyTypeToDisplay.Value = StoryElementType.Outro;
  }

  @bind
  onBackToSelectionButtonClicked(): void {
    this.viewModel.storyTypeToDisplay.Value = StoryElementType.IntroOutro;
  }
}
