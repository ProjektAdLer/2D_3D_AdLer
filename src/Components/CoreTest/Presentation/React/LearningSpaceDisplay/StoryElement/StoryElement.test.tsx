import { fireEvent, render } from "@testing-library/react";
import { mock } from "jest-mock-extended";
import React from "react";
import StoryElement from "../../../../../Core/Presentation/React/LearningSpaceDisplay/StoryElement/StoryElement";
import StoryElementController from "../../../../../Core/Presentation/React/LearningSpaceDisplay/StoryElement/StoryElementController";
import StoryElementViewModel from "../../../../../Core/Presentation/React/LearningSpaceDisplay/StoryElement/StoryElementViewModel";
import useBuilderMock from "../../ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import { StoryElementType } from "../../../../../Core/Domain/Types/StoryElementType";
import { LearningElementModelTypeEnums } from "../../../../../Core/Domain/LearningElementModels/LearningElementModelTypes";

let viewModel = new StoryElementViewModel();
viewModel.isOpen.Value = true;
const fakeController = mock<StoryElementController>();

describe("StoryElement", () => {
  beforeEach(() => {
    viewModel.introTexts.Value = ["Blabla Intro 1"];
    viewModel.outroTexts.Value = ["Blabla Outro 1"];
    viewModel.isOpen.Value = true;
    viewModel.isOutroUnlocked.Value = false;
    viewModel.isIntroCutsceneRunning.Value = false;
    viewModel.isOutroCutsceneRunning.Value = false;
    viewModel.storyTypeToDisplay.Value = StoryElementType.None;
  });

  test("doesn't render without controller", () => {
    useBuilderMock([viewModel, undefined]);
    const { container } = render(<StoryElement />);
    expect(container.firstChild).toBeNull();
  });

  test("doesn't render without view model", () => {
    useBuilderMock([undefined, fakeController]);
    const { container } = render(<StoryElement />);
    expect(container.firstChild).toBeNull();
  });

  test("doesn't render when storyTypeToDisplay is None", () => {
    viewModel.storyTypeToDisplay.Value = StoryElementType.None;
    useBuilderMock([viewModel, fakeController]);
    const { container } = render(<StoryElement />);
    expect(container.firstChild).toBeNull();
  });

  // ANF-ID: [ELG0032]
  test.each([
    [LearningElementModelTypeEnums.QuizElementModelTypes.ArcadeNPC],
    [LearningElementModelTypeEnums.QuizElementModelTypes.CampusNPC],
    [LearningElementModelTypeEnums.QuizElementModelTypes.DefaultNPC],
    [LearningElementModelTypeEnums.QuizElementModelTypes.RobotNPC],
  ])("displays correct npc thumbnail for model %p", (model) => {
    viewModel.introModelType.Value = model;
    viewModel.isOpen.Value = true;
    viewModel.storyTypeToDisplay.Value = StoryElementType.Intro;
    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);

    const npcThumbnails = componentUnderTest.getAllByTestId("npcImage");

    npcThumbnails.forEach((npcThumbnail) => {
      expect(npcThumbnail).toBeInTheDocument();
      expect(npcThumbnail).toMatchSnapshot();
    });
  });

  // ANF-ID: [EWE0041]
  test("should not render when closed", () => {
    viewModel.isOpen.Value = false;

    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);
    expect(componentUnderTest.container.childElementCount).toBe(0);
  });

  // ANF-ID: [EWE0041]
  test("should close when x Button is clicked in story selection", () => {
    viewModel.isOpen.Value = true;
    viewModel.storyTypeToDisplay.Value = StoryElementType.IntroOutro;

    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);
    const xButton = componentUnderTest.getByAltText("CloseButton");
    fireEvent.click(xButton);
    expect(fakeController.closePanel).toHaveBeenCalled();
  });

  //ANF-ID: [EWE0040]
  test("should render with introTitle and introTexts when storyTypeToDisplay is Intro", () => {
    viewModel.storyTypeToDisplay.Value = StoryElementType.Intro;
    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);
    const title = componentUnderTest.getByText("introStoryTitle");
    const text = componentUnderTest.getByText("Blabla Intro 1");

    expect(title).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });

  //ANF-ID: [EWE0040]
  test("should render with outroTitle and outroTexts when storyTypeToDisplay is Outro and isOutroUnlocked is true", () => {
    viewModel.storyTypeToDisplay.Value = StoryElementType.Outro;
    viewModel.isOutroUnlocked.Value = true;
    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);
    const title = componentUnderTest.getByText("outroStoryTitle");
    const text = componentUnderTest.getByText("Blabla Outro 1");

    expect(title).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });

  //ANF-ID: [EWE0040]
  test("should render with outroTitle and outroLockedText placeholder when storyTypeToDisplay is Outro and isOutroUnlocked is false", () => {
    viewModel.storyTypeToDisplay.Value = StoryElementType.Outro;
    viewModel.isOutroUnlocked.Value = false;
    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);
    const title = componentUnderTest.getByText("outroStoryTitle");
    const text = componentUnderTest.getByText("outroLockedText");

    expect(title).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });

  //ANF-ID: [EWE0040]
  test("should render selection when storyTypeToDisplay is IntroOutro", () => {
    viewModel.storyTypeToDisplay.Value = StoryElementType.IntroOutro;
    useBuilderMock([viewModel, fakeController]);

    const componentUnderTest = render(<StoryElement />);
    const introButton = componentUnderTest.getByRole("button", {
      name: "introStoryTitle",
    });
    const outroButton = componentUnderTest.getByRole("button", {
      name: "outroStoryTitle",
    });

    expect(introButton).toBeInTheDocument();
    expect(outroButton).toBeInTheDocument();
  });
});
