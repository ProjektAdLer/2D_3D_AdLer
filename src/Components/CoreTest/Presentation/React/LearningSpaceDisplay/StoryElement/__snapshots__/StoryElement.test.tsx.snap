// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryElement displays correct npc thumbnail for model "a-npc-alerobot" 1`] = `
<img
  alt="LearningImage!"
  class="invisible z-20 h-0 -scale-x-100 object-contain brightness-125 lg:visible lg:h-full"
  data-testid="npcImage"
  src="a-npc-alerobot-welcome.png"
/>
`;

exports[`StoryElement displays correct npc thumbnail for model "a-npc-defaultarcade" 1`] = `
<img
  alt="LearningImage!"
  class="invisible z-20 h-0 -scale-x-100 object-contain brightness-125 lg:visible lg:h-full"
  data-testid="npcImage"
  src="a-npc-sheriffjustice-welcome.png"
/>
`;

exports[`StoryElement displays correct npc thumbnail for model "a-npc-defaultcampus" 1`] = `
<img
  alt="LearningImage!"
  class="invisible z-20 h-0 -scale-x-100 object-contain brightness-125 lg:visible lg:h-full"
  data-testid="npcImage"
  src="a-npc-teamb-male-welcome.png"
/>
`;

exports[`StoryElement displays correct npc thumbnail for model "a-npc-defaultsuburb" 1`] = `
<img
  alt="LearningImage!"
  class="invisible z-20 h-0 -scale-x-100 object-contain brightness-125 lg:visible lg:h-full"
  data-testid="npcImage"
  src="a-npc-teamd-male-welcome.png"
/>
`;
